apt-hook/esm-counts.cc

```diff
@@ -52,11 +52,26 @@ bool get_potential_esm_updates(ESMUpdates &updates) {
    }
 
    // set up esm cache
+   // make sure the configuration is isolated, apart from Acquire
+   const Configuration::Item *ConfigItem = _config->Tree(0);
+   while (ConfigItem) {
+      if (ConfigItem->FullTag(0) != "Acquire") {
+         _config->Clear(ConfigItem->FullTag(0));
+      }
+      ConfigItem = ConfigItem->Next;
+   }
```

Comment:
Optional: I think we could turn this into a for loop to be maybe slightly more readable. Something like this:
```suggestion
   for (Configuration::Item *ConfigItem = _config->Tree(0); ConfigItem != NULL; ConfigItem = ConfigItem->Next) {
      if (ConfigItem->FullTag(0) != "Acquire") {
         _config->Clear(ConfigItem->FullTag(0));
      }
   }
```

Reply:
sure why not